cpp/tests/reductions/scan_tests.cpp

```diff
@@ -14,584 +14,390 @@
-  if (do_print) {
-    print_view(expected_col_out1, "expect = ");
-    print_view(col_out->view(), "result = ");
+  std::unique_ptr<cudf::column> make_expected(cudf::host_span<HostType const> v,
+                                              cudf::host_span<bool const> b,
+                                              std::unique_ptr<aggregation> const& agg,
+                                              scan_type inclusive,
+                                              null_policy null_handling,
+                                              numeric::scale_type scale = numeric::scale_type{0})
+  {
+    auto op       = this->make_agg(agg);
+    auto identity = this->make_identity(agg);
+
+    thrust::host_vector<HostType> expected(v.size());
+    thrust::host_vector<bool> b_out(b.begin(), b.end());
+
+    bool const nullable = (b.size() > 0);
+
+    auto masked_value = [identity](auto const& z) {
+      return thrust::get<1>(z) ? thrust::get<0>(z) : identity;
+    };
+
+    if (inclusive == cudf::scan_type::INCLUSIVE) {
+      if (nullable) {
+        std::transform_inclusive_scan(
+          thrust::make_zip_iterator(thrust::make_tuple(v.begin(), b.begin())),
+          thrust::make_zip_iterator(thrust::make_tuple(v.end(), b.end())),
+          expected.begin(),
+          op,
+          masked_value);
+
+        if (null_handling == null_policy::INCLUDE) {
+          std::inclusive_scan(b.begin(), b.end(), b_out.begin(), std::logical_and<bool>{});
+        }
+      } else {
+        std::inclusive_scan(v.begin(), v.end(), expected.begin(), op);
+      }
+    } else {
+      if (nullable) {
+        std::transform_exclusive_scan(
+          thrust::make_zip_iterator(thrust::make_tuple(v.begin(), b.begin())),
+          thrust::make_zip_iterator(thrust::make_tuple(v.end(), b.end())),
+          expected.begin(),
+          identity,
+          op,
+          masked_value);
+
+        if (null_handling == null_policy::INCLUDE) {
+          std::exclusive_scan(b.begin(), b.end(), b_out.begin(), true, std::logical_and<bool>{});
+        }
+      } else {
+        std::exclusive_scan(v.begin(), v.end(), expected.begin(), identity, op);
+      }
+    }
+
+    return nullable ? this->make_column(expected, b_out, scale)
+                    : this->make_column(expected, {}, scale);
   }
 
-  // skipna=false
-  CUDF_EXPECT_NO_THROW(
-    col_out =
-      cudf::scan(col_in, cudf::make_sum_aggregation(), scan_type::INCLUSIVE, null_policy::INCLUDE));
-  cudf::test::fixed_width_column_wrapper<TypeParam> expected_col_out2(
-    out_v.begin(), out_v.end(), out_b.begin());
-  if (do_print) {
-    print_view(expected_col_out2, "expect = ");
-    print_view(col_out->view(), "result = ");
+  template <typename Ti>
+  void print(thrust::host_vector<Ti> const& v, const char* msg = nullptr)
+  {
+    std::cout << msg << " {";
+    std::for_each(v.begin(), v.end(), [](Ti i) { std::cout << ", " << i; });
```

Comment:
There's no way around that leading `,`, is there?

Reply:
This has all been eliminated and simplified. Thanks for catching.